src/base/bittorrent/session.cpp

```diff
@@ -1863,20 +1863,26 @@ TorrentHandle *Session::findTorrent(const InfoHash &hash) const
 
 bool Session::hasActiveTorrents() const
 {
-    foreach (TorrentHandle *const torrent, m_torrents)
-        if (TorrentFilter::ActiveTorrent.match(torrent))
-            return true;
-
-    return false;
+    return std::any_of(m_torrents.begin(), m_torrents.end(), [](TorrentHandle *torrent)
```

Comment:
Why does the pointer to a non-constant object?

Reply:
ActiveTorrent.match() takes a non-const, and I didn't want to risk breaking a bunch of other things by changing it.

Reply:
Fine then, but there are similar lambdas below. Are non-const object needed there too?

Reply:
No, not particularly.

Reply:
Would you, please, add `const` to their parameters?

Reply:
k

Reply:
thank you.